### PR TITLE
Accelerator profile table filter

### PR DESCRIPTION
**Pull request by dpanshug**
Closes: #1374

## Description
Added filter to accelerator profile table.

Screenshot from 2023-11-16 02-02-18

Screenshot from 2023-11-16 02-01-55

## How Has This Been Tested?
1. Add accelerator profile CR.
2. Settings < accelerator profiles
3. On accelerator profile table page, try filter some accelerators.

## Test Impact
Added some integration test within List accelerator profile tests.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

@yannnz

### CODE REVIEW

**Comment by dpanshug** [comment]
Upon assigning Gage as the reviewer, other review requests are removed automatically.

**Review by Gkrumbach07 — APPROVED** [review]
/lgtm

Tested and works

**Comment by Gkrumbach07** [comment]
@yannnz pending a quick review from you on the UX

File: frontend/src/pages/acceleratorProfiles/screens/list/AcceleratorProfilesTable.tsx [hunk 1]
```diff
@@ -12,26 +14,63 @@ type AcceleratorProfilesTableProps = {
...
 
   return (
     <Table
       id="accelerator-profile-table"
       enablePagination
-      data={accelerators}
+      data={filteredAcceleratorProfiles}
       columns={columns}
-      emptyTableView={'No projects match your filters.'}
+      emptyTableView={<DashboardEmptyTableView onClearFilters={resetFilters} />}
```
Comment (manaswinidas) [inline frontend/src/pages/acceleratorProfiles/screens/list/AcceleratorProfilesTable.tsx:46]: Can you inline this? Just like what you've done for `onSearchTypeChange` and `onSearchValueChange`.

**Review by manaswinidas — CHANGES_REQUESTED** [review]

**Review by manaswinidas — COMMENTED** [review]
/lgtm

**Comment by yannnz** [comment]
> @yannnz pending a quick review from you on the UX

Looks good. Thanks. @dpanshug @Gkrumbach07

**Pull request merged**